pygmt/clib/conversion.py

```diff
@@ -103,7 +103,7 @@ def dataarray_to_matrix(grid):
                 "but GMT only supports regular spacing. Calculated regular spacing "
                 f"{coord_inc} is assumed in the '{dim}' dimension."
             )
-            warnings.warn(msg, category=RuntimeWarning)
+            warnings.warn(msg, category=RuntimeWarning, stacklevel=1)
```

Comment:
Should this be `stacklevel=2` instead? We're using `stacklevel=2` in most other places, e.g. https://github.com/GenericMappingTools/pygmt/blob/c40e785371bfb34208604d2a248d4508a12b8cb3/pygmt/clib/session.py#L1575

and from a quick search (e.g. https://github.com/python/cpython/issues/88462), it seems like `stacklevel=2` is recommended for warnings?
```suggestion
            warnings.warn(msg, category=RuntimeWarning, stacklevel=2)
```